### PR TITLE
🏗 Build Spider: Cincinnati Metro Housing Authority

### PR DESCRIPTION
**Pull request by cruznunez**
## What's this PR do?
This PR adds a scraper for Cincinnati Metro Housing Authority Commission. The scraper fetches meetings from 6 months ago to 6 months in the future via an API endpoint. Usually there are not many meetings in the future.

## Why are we doing this?
Scraper requested from [spreadsheet](https://docs.google.com/spreadsheets/d/1-HwPQKwm56vZ8h7CEM65GazKLTS0086gmaEzEHXp7O4/edit?usp=sharing).

## Steps to manually test
1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spider:
```
scrapy crawl caconj_metro_housing_authority -O test_output.csv
```
4. Monitor the output and ensure no errors are raised.
5. Inspect `test_output.csv` to ensure the data looks valid.
6. Ensure all tests pass
```
pytest
```

## Are there any smells or added technical debt to note?
No

## Summary by CodeRabbit

- **New Features**
	- Added a new spider to scrape meeting information for the Cincinnati Metro Housing Authority.
	- Implemented parsing of calendar events with detailed metadata, including status and links.
	- Introduced a JSON data structure with multiple event objects for upcoming meetings.

- **Tests**
	- Created a comprehensive test suite for the new spider.
	- Verified parsing of meeting details, statuses, locations, and links, ensuring accuracy and consistency.

### CODE REVIEW

File: city_scrapers/spiders/caconj_metro_housing_authority.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        """
+        Parse API response from Executive Office cintimha.com calendar.
+        """
+
+        data = response.json()
+
+        for item in data:
+            meeting = Meeting(
+                title=item["title"],
+                description="",
```
[the remark this answers is not in the document]
Comment (cruznunez) [reply city_scrapers/spiders/caconj_metro_housing_authority.py:41]: I used the API description for the meeting description at the beginning, but when I ran
```bash
scrapy crawl caconj_metro_housing_authority -O test_output.csv
```
I was getting strange syntax highlighting for the CSV. I assumed it was malformed CSV. It went away when I removed the API description. See below.

 

Is this okay or do we have to do something else?

File: city_scrapers/spiders/caconj_metro_housing_authority.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                location["address"] = ", ".join(parts[1:])
+        return location
+
+    def _parse_links(self, item):
+        """Parse or generate links."""
+        return [{"title": "Event Link", "href": item["url"]}]
+
+    def _parse_source(self, response):
+        """Parse or generate source."""
+        return response.url
```
[the remark this answers is not in the document]
Comment (cruznunez) [reply city_scrapers/spiders/caconj_metro_housing_authority.py:97]: Yes. That makes sense. Thank you.

File: city_scrapers/spiders/caconj_metro_housing_authority.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        end = item["end"]
+        end_no_tz = parse(end, ignoretz=True)
+        return end_no_tz
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
+        location = {"name": "", "address": ""}
+        text = item["description"]
+        regex = r"(at)?.+\s(at|the)\s?(.+)\sbeginning"
+        match = re.search(regex, text)
```
[the remark this answers is not in the document]
Comment (cruznunez) [reply city_scrapers/spiders/caconj_metro_housing_authority.py:74]: Good catch. I also don't get any locations scraped even though it works in the tests. Will fix. Thanks!

[the remark this answers is not in the document]
Comment (cruznunez) [reply city_scrapers/spiders/caconj_metro_housing_authority.py]: I think one underlying problem to this is that the URL generated by the spider before scraping was incomplete. 

```python
# before
url = "https://cintimha.com/wp-admin/admin-ajax.php?action=get_calendar_events"
        f"&noheader=true&start_date={start_int}&end_date={end_int}"
        "&show_expired=true&event_category_id=executive-office-1466450825"
# after
url = (
            "https://cintimha.com/wp-admin/admin-ajax.php?action=get_calendar_events"
            f"&noheader=true&start_date={start_int}&end_date={end_int}"
            "&show_expired=true&event_category_id=executive-office-1466450825"
        )
```
It was just using the first line of the above string. 

It was 
```ruby
https://cintimha.com/wp-admin/admin-ajax.php?action=get_calendar_events
```
instead of 
```ruby
https://cintimha.com/wp-admin/admin-ajax.php?action=get_calendar_events
&noheader=true&start_date={start_int}&end_date={end_int}
&show_expired=true&event_category_id=executive-office-1466450825
```

This was fetching the wrong events and was not using the correct time period either. Fixing that makes the spider fetch the correct events and all locations are scraped.

**Review by lamle-ea — COMMENTED** [review]
Hi @cruznunez, I've noticed an issue with the event filtering. When running the scraper with `event_category_id=executive-office-1466450825` in the URL, it only returns meetings from 2024. However, removing this parameter successfully retrieves 2025 meetings.

Could you explain the purpose of including this category ID parameter? It seems to be limiting our results unexpectedly.

**Comment by cruznunez** [comment]
@lamle-ea in the notes column of the spreadsheet it says
> Note - the meetings we want to scraper are under the category Executive office.

On the calendar page, when I click on Category dropdown and select Executive Office, I see the webpage make a request to 
```ruby
https://cintimha.com/wp-admin/admin-ajax.php?action=get_calendar_events&noheader=true&start_date=1735448400&end_date=1738472400&show_expired=true&event_category_id=executive-office-1466450825&event_venue_id=&iframe=
```
This is the URL that got adapted into
```python
f"https://cintimha.com/wp-admin/admin-ajax.php?action=get_calendar_events&noheader=true&start_date={start_int}&end_date={end_int}&show_expired=true&event_category_id=executive-office-1466450825"
```

**Comment by lamle-ea** [comment]
Thanks for confirming. I did not notice the note in the spreadsheet.

**Review by lamle-ea — APPROVED** [review]
LGTM

**Comment by cruznunez** [comment]
Looks like there has been a mistake. The Cincinnati Metro Housing Authority spider should be in the [Cincinnati, OH](https://github.com/City-Bureau/city-scrapers-cinoh/tree/main) repo, not the [Cape May County, NJ](https://github.com/City-Bureau/city-scrapers-caconj/tree/main) repo. Will revert and make a PR to the appropriate repo.
